internal/controller/platform/standalone/nimservice.go

```diff
@@ -372,6 +465,55 @@ func (r *NIMServiceReconciler) getNIMCachePVC(ctx context.Context, nimService *a
 	return &nimCache.Spec.Storage.PVC, nil
 }
 
+func (r *NIMServiceReconciler) getTensorParallelismByProfile(ctx context.Context, profile *appsv1alpha1.NIMProfile) (string, error) {
+	tensorParallelism := ""
+	if tp, exists := profile.Config["tp"]; exists {
+		tensorParallelism = tp
+	}
+
+	return tensorParallelism, nil
+}
+
+func (r *NIMServiceReconciler) getDeviceIDByProfile(ctx context.Context, profile *appsv1alpha1.NIMProfile) (string, error) {
+	deviceID := ""
+	if device, exists := profile.Config["gpu_device"]; exists {
```

Comment:
from the model point of view, the same (model, profile) should be able to run on multiple types of gpus. But here the logic seem to indicate that it is always the 1:1 mapping. Does nim enforce the latter?

Reply:
@slu2011 models are optimized for specific GPU SKUs, they cannot be run on other types.

Reply:
How about different mig profiles from the same physical gpu? like 1g.10gb vs  2g.20gb. Does nim provide different model even for different mig profiles, or it would be the same?

Reply:
@slu2011 currently NIMs need a full GPU, they don't support MIG.